examples/cpp/hello_world/main.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file main.cpp
+ *
+ */
+
+#include <csignal>
+#include <stdexcept>
+#include <thread>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/log/Log.hpp>
+
+#include "Application.hpp"
+#include "CLIParser.hpp"
+
+using eprosima::fastdds::dds::Log;
+
+using namespace eprosima::fastdds::examples::hello_world;
+
+std::function<void(int)> stop_app_handler;
+void signal_handler(
+        int signum)
+{
+    stop_app_handler(signum);
+}
+
+int main(
+        int argc,
+        char** argv)
+{
+    auto ret = EXIT_SUCCESS;
+    const std::string topic_name = "hello_world_topic";
+    CLIParser::hello_world_config config = CLIParser::parse_cli_options(argc, argv);
+    uint16_t samples = config.entity == CLIParser::EntityKind::PUBLISHER ? config.pub_config.samples :
+            config.entity == CLIParser::EntityKind::SUBSCRIBER ? config.sub_config.samples : 0;
```

Comment:
I think two ternary operators make the code difficult to follow. I think an `if-else if-else` would be more readable